ar_detection.py

```diff
@@ -518,17 +520,89 @@ def create_shapefile(all_ars, shp_fp, csv_fp):
     pd.DataFrame.from_dict(data=col_dict, orient='index').to_csv(csv_fp, header=False)
 
 
-def detect_all_ars(fp, n_criteria, out_shp, out_csv):
+def landfall_ars_export(shp_fp, ak_shp, fp_6hr, fp_events):
+    """Filter the raw AR detection shapefile output to include only ARs making landfall in Alaska, and export the result to a new shapefile. Process the landfall ARs to condense adjacent dates into multipolygons, and export the result to a second new shapefile.
+
+    Parameters
+    ----------
+    shp_fp : PosixPath
+        File path to the raw AR detection shapefile input.
+    ak_shp : PosixPath
+        File path to the Alaska coastline shapefile input.
+    fp_6hr : PosixPath
+        File path for the raw 6hr interval landfall AR shapefile output.
+    fp_events : PosixPath
+        File path for the condensed landfall AR events shapefile output.
+
+    Returns
+    -------
+    None
+    """
+
+    # import AK coastline and AR shps to gdfs
+    ars = gpd.read_file(shp_fp)
+    ak = gpd.read_file(ak_shp)
+
+    # reproject AK coastline to match AR CRS, and dissolve into one multipolygon
+    ak_ = ak.to_crs(ars.crs)
+    ak_d = ak_.dissolve()
+
+    # add new datetime column to ars gdf by parsing ISO timestamp
+    # reformat time column string for output (datetime fields not supported in ESRI shp files)
+    ars['dt'] = [datetime.fromisoformat(ars.time.iloc[[d]].values[0][:-16]) for d in range(len(ars))]
+    ars['time'] = ars['dt'].astype(str)
+
+    # perform spatial join, keeping only AR polygons that intersect with the AK polygon
+    ak_ars = ars.sjoin(ak_d, how='inner', predicate='intersects')
+
+    # export raw 6hr geodataframe to shp
+    ak_ars.drop(columns=['dt', 'index_right', 'FEATURE']).to_file(fp_6hr, index=True)
+
+    # label any ARs occuring on adjacent dates with a unique "diff" ID
+    # for each ID, turn first and last timestep into gdf attributes and combine their geometry into one multipolygon feature
+
+    ak_ars['diff'] = ak_ars['dt'].diff().dt.days.gt(1).cumsum()
+
+    dfs=[]
```

Comment:
```suggestion
    dfs = []
```
Run the Black formatter or just check for single whitespace around these operators, ditto for the single vs. double quote encapsulation in this function.